.github/workflows/rust-pipeline.yml

```diff
@@ -0,0 +1,101 @@
+name: Rust pipeline
```

Comment:
Might be overly pedantic but I think this should be called rust workflow, since it's a called workflow. I believe several workflows make a pipeline in github terms